lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java

```diff
@@ -110,18 +112,10 @@ private void check(DOMDocument document) {
 		if (lastKnownVersionMavenProject != null) {
 			return Optional.of(lastKnownVersionMavenProject);
 		}
-		if (mavenRequest == null) {
```

Comment:
@mickaelistria this is the line that made me assume the maven build state shouldn't be initialized if the mavenRequest is non-null. 

I assume the way I have it currently is ok? Ie. don't initialize the Maven Build State if the mavenRequest is null.

Reply:
I think we should care about the mavenRequest at this point.

Reply:
So do you think the maven build state should still be initialized in all areas as previously? Or only in the constructor is OK? (as I have it currently)